frontend/packages/console-plugin-sdk/src/typings/overview.ts

```diff
@@ -0,0 +1,50 @@
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { FirehoseResource } from '@console/internal/components/utils';
+import { OverviewDetailsResourcesTabProps } from '@console/internal/components/overview/resource-overview-page';
+import { OverviewMainContentProps } from '@console/internal/components/overview';
+import { Extension } from './extension';
+import { LazyLoader } from './types';
+
+type ResourceItem = {
+  [key: string]: K8sResourceKind[];
+};
+
+namespace ExtensionProperties {
+  export interface OverviewCRD {
+    /** Resources list to be fetched from Firehose. */
+    resources: (namespace: string) => FirehoseResource[];
+
+    /** The name of the feature flag. */
+    required: string;
+
+    /** util to check get resources. */
+    utils: (dc: K8sResourceKind, props: OverviewMainContentProps) => ResourceItem;
+  }
+
+  export interface OverviewResourceTab {
+    /** The name of Overview tab to be updated. */
+    name: string;
+
+    /** Name of key to be checked in prop items. */
+    key: string;
+
+    /** Loader for the corresponding tab component. */
+    loader: LazyLoader<OverviewDetailsResourcesTabProps>;
```

Comment:
I feel like the props definition is backwards here. I think nav does it this way too so something we can fix later. I feel like the plugin sdk should define the props and the component should use the sdk props. Not the other way around. Cc @vojtechszocs